common/persistence/visibility/store/standard/cassandra/visibility_store.go

```diff
@@ -460,8 +437,47 @@ func (v *visibilityStore) ListClosedWorkflowExecutionsByStatus(
 	return response, nil
 }
 
-// DeleteWorkflowExecution is a no-op since deletes are auto-handled by cassandra TTLs
-func (v *visibilityStore) DeleteWorkflowExecution(_ *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+func (v *visibilityStore) DeleteWorkflowExecution(request *manager.VisibilityDeleteWorkflowExecutionRequest) error {
+	// Primary key in closed_execution table has close_time which is not in the request.
+	// Query closed_execution table first (using secondary index by workflow_id) to get close_time.
+	// This is not very efficient if workflow has multiple executions.
+	query := v.session.
+		Query(templateGetClosedWorkflowExecutionsByID,
+			request.NamespaceID.String(),
+			namespacePartition,
+			persistence.UnixMilliseconds(minTime),
+			persistence.UnixMilliseconds(maxTime),
+			request.WorkflowID).
+		Consistency(v.lowConslevel)
+	iter := query.Iter()
+
+	wfExecution, has := readClosedWorkflowExecutionRecord(iter)
```

Comment:
Does this query handles pagination as well? If workflowID is a cron, you could have many RunIDs.

Reply:
Not relevant anymore, but it turned out that default page size is 5000 (not unlimited, as I would expect it to be if not specified).